src/render/mesh.cpp

```diff
@@ -338,7 +338,7 @@ MI_VARIANT void Mesh<Float, Spectrum>::recompute_vertex_normals() {
         // --------------------- Kernel 2 starts here ---------------------
 
         normals = dr::normalize(normals);
-
+        m_vertex_normals = dr::detach(m_vertex_normals);
```

Comment:
```suggestion
       dr::disable_grad(m_vertex_normals);
```

Reply:
You are right, using disable_grad is better here. Also added a comment explaining why its needed